spring/src/main/java/org/axonframework/spring/authorization/AuthorizationMessageDispatchInterceptor.java

```diff
@@ -0,0 +1,41 @@
+package org.axonframework.spring.authorization;
+
+import org.axonframework.commandhandling.CommandMessage;
+import org.axonframework.messaging.MessageDispatchInterceptor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BiFunction;
+
+/**
+ * Message dispatch interceptor that adds the {$code username} and {$code authorities} from the authorized principle
+ *
+ * @author Roald Bankras
+ */
+public class AuthorizationMessageDispatchInterceptor implements MessageDispatchInterceptor<CommandMessage<?>> {
```

Comment:
Although commands are arguable the biggest entry point from the outside world into an Axon Framework application, I think it would be nice if we could use this interceptor for the `QueryMessage` too.

Since the `QueryMessage` is just another message that would originate from "the outside world", the same Spring Security logic can take place on those.

So, @bankras, would you mind adjusting the interceptor to accept the generic `? extends Message<?>`, instead of the `CommandMessage<?>`?